ecl/hql/hqlgram.y

```diff
@@ -5986,6 +5987,23 @@ primexpr1
                             parser->normalizeExpression($7, type_int, false);
                             $$.setExpr(createValue(no_regex_find, subType.getLink(), $3.getExpr(), $5.getExpr(), $7.getExpr(), $8.getExpr()));
                         }
+    | REGEXFINDSET '(' expression ',' expression regexOpt ')'
+                        {
+                            parser->normalizeExpression($3, type_stringorunicode, false);
+                            Owned<ITypeInfo> retType;
+                            if(isUnicodeType($3.queryExprType()))
+                            {
+                                retType.setown(makeUnicodeType(UNKNOWN_LENGTH, 0));
```

Comment:
minor: Better as makeUnicodeType(UNKNOWN_LENGTH, $3.queryExprType()->queryLocale())

Yes, the existing regexfind and replace are also wrong (and should really have been passing NULL instead of 0...)